Language/Haskell/GhcMod/Target.hs

```diff
@@ -188,12 +187,22 @@ targetGhcOptions crdl sefnmn = do
     when (Set.null sefnmn) $ error "targetGhcOptions: no targets given"
 
     case cradleProject crdl of
-      proj
-          | isCabalHelperProject proj -> cabalOpts crdl
-          | otherwise -> sandboxOpts crdl
+      proj | isCabalHelperProject proj -> cabalOpts crdl
+      ExplicitProject -> do
+        opts <- explicitOpts crdl
+        dbStack <- maybe [] id <$> getCustomPkgDbStack
+        return $ opts ++ ghcDbStackOpts dbStack
+      _  -> sandboxOpts crdl
  where
    zipMap f l = l `zip` (f `map` l)
 
+   explicitOpts :: Cradle -> GhcModT m [String]
+   explicitOpts Cradle {..} = case cradleCabalFile of
+       Nothing -> return []
```

Comment:
Could you explain what you're doing here? Looking at what `explicitCradle` above it seems that when the project type is ExplicitCradle """cradleCabalFile""" would always be: `cradleCabalFile  = Just optionsFile` which seems like a horrible misnomer in its own right ;)